### DISCUSSION TITLE
Code/Monospaced text has light theme background, even in dark theme

### DISCUSSION THREAD

**Discussion by moonpiedumplings** [quarto-dev/quarto-cli#9237] [Q&A]
### Description

I'm not too knowledgeable about frontend, so I haven't investigated the theming and such too closely. 

But the gist is that monospaced text, the one delimited by single backticks (like so): 

```
`single backtick`
```

On light mode, it renders properly, but the 

But here is an example for my site: 

https://moonpiedumplings.github.io/playground/ova-automation/

On this site, code blocks remain light while in light mode... and in dark mode.

Here is my _quarto.yml again: 

```{.yaml}
project:
  type: website
  render:
    - "*.*md"
    - "!scrapyard/"
    - "!not*"
    - "!norender*"

website:
  title: "Jeffrey Fonseca"
  site-url: https://moonpiedumplings.github.io
  navbar:
    left:
      - projects/index.qmd
      - blog/index.qmd
      - playground/index.qmd
      - guides/index.qmd
      - talks/index.qmd
      - writeups/index.qmd
      - text: Resume
        href: resume/index.qmd
      #- about
    right:
      - icon: github
        href: https://github.com/moonpiedumplings
format:
  html:
    theme:
      light: flatly
      dark: darkly
    css: styles.css
    toc: true
    anchor-sections: true
```

I think it could be related to this issue: https://github.com/quarto-dev/quarto-cli/issues/8568

But I don't understand much of what is going on there, and it seems to still be open.

In addition to that my `styles.css` file is completely empty, only containing the autogenerated content when you first create a quarto website. 

```{.css}
/* css styles */
```

And here is the git repo: https://github.com/moonpiedumplings/moonpiedumplings.github.io/

Does anyone have any idea on what could be causing this?

This started at around quarto 1.3, and I haven't been able to figure out anything since then. I think this is a bug. 

EDIT:

Yeah, I think this is a bug.  A quick test with an older version of quarto (1.2.335) and it gives me the correct coloring. I used nix to get an older version:

`nix-shell -p quarto -I nixpkgs=https://github.com/NixOS/nixpkgs/archive/8ad5e8132c5dcf977e308e7bf5517cc6cc0bf7d8.tar.gz`

**Comment by mcanouil** [comment]
Unfortunately, your example is far from small for us to investigate.
Could you try to make it small, for example starting with a default website and adding only what is needed to reproduce.

```sh
quarto create project website discussion-9237 Discussion-9237
```

This being said, code highlighting is defined by theme that can be changed, see <https://quarto.org/docs/output-formats/html-code.html#highlighting>.
There is currently no Quarto YAML option to set colour for this, but it can be controlled using (S)CSS. To do so, take a look at [HTML theming documentation](https://quarto.org/docs/output-formats/html-themes.html#sass-variables) and use your favourite browser inspect/developer mode.

**Comment by cderv** [comment]
> [moonpiedumplings.github.io/playground/ova-automation](https://moonpiedumplings.github.io/playground/ova-automation/)

> On this site, code blocks remain light while in light mode... and in dark mode.

Is this still the case ? I can't seem to reproduce this when looking at your website on Chrome. 
![chrome_5Oof1WI7yb](https://github.com/quarto-dev/quarto-cli/assets/6791940/9a2fa554-a5f8-4541-a9dc-45aad16d07e7)

**Reply by moonpiedumplings** [reply]
Yeah, I found a workaround. Following the theming guides, I changed the code block colors. 

From: https://github.com/moonpiedumplings/moonpiedumplings.github.io/blob/b3b2e9951231849ad068aca82c24eff9940c6462/_quarto.yml

```{.yaml}

format:
  html:
    theme:
      light: 
        - light-styles.scss
        - flatly 
      dark: 
        - dark-styles.scss
        - darkly
``

And in `dark-styles.scss` ([linked here](https://github.com/moonpiedumplings/moonpiedumplings.github.io/blob/b3b2e9951231849ad068aca82c24eff9940c6462/dark-styles.scss)

```{.scss}
/*-- scss:defaults --*/

$code-color: #9753b8;
$code-bg: #2b2b2b;
```

And with this, code blocks look dark again. This still seems like a workaround, and I don't think the bug is fixed properly, but it works for now.
